### ISSUE TITLE
Add release channels to auto updater

### ISSUE THREAD

**Issue by Daenyth** [Cockatrice/Cockatrice#2321]
Currently if we host a build, the auto updater will download it. This means we can't effectively release beta or RC versions without having everyone start running it. We should add in release channels so that people who want to get beta versions will get them installed by the updater, and people who want stable can stay on full releases.

To start:
- Stable (our current releases and release process)
- Beta (manually released in the same way stable is, but not full releases)
- Unstable (nightly builds from master branch)

**Comment by ZeldaZach** [comment]
I fully agree with this, and I'd love it if we can prioritize this for the release scheduled at the end of month. Beta/Unstable is a bit weird for us as we have 2 build channels; One that is manually updated by me with binaries, and one that is auto built every time a PR is merged. So we'd have to change the logic behind how stable vs beta is determined (right now, the way it parses bintray, i need to delete old builds for the updater to detect a change) then I'm all for it and we can easily sustain those 3 types of channels.

**Comment by Daenyth** [comment]
I don't think this will get finished before then.

I don't think this is [API] since there's no server-side component here.

**Comment by tooomm** [comment]
I like that! 👍 

Right now we have on bintray:
`cockatrice`: manually uploaded releases --> stable
`cockatrice-git`: automatic builds of master --> unstable

So we can go with these two at first maybe and extend the updater to offer downloads of unstable builds.

For the beta channel... would it be possible to separate the manual uploads in `cockatrice` like that:
"old/current releasename" --> stable
"new releasename + beta tag in name" --> beta
Once the beta gets stable, the beta tag is removed and the new release with its new name will get the beta label.
Then it might be possible to separate the two versions by two different queries.
If we separate both with an additional layer of folders (stable + beta) old clients with their update structure won't work any longer I guess?

**On the other hand, I'm not sure if we should offer unstable builds like that in-client at all. It adds a lot of traffic which might be problematic with bintray. And if people want to run experimental builds, they should deal with the effort of manually installing them. If they can't do that, they are most likely the first to complain and post issues about problems which are not too helpful just because they don't understand the meaning of unstable software in the first place.**

**Comment by Daenyth** [comment]
It's reasonable to not auto-update unstable builds.

**Comment by tooomm** [comment]
Oh, we are really speaking about real auto-updating?
Since we don't have that at all right now... all manual updates as of now. No automatic check and download.

**Comment by Daenyth** [comment]
I thought the windows updater checked for new versions on startup?

**Comment by tooomm** [comment]
As far as I know it just checks on demand. (Help --> Update Cockatrice)
It should be the same on all OS. There's also no setting to influence this behavior!

**Comment by ZeldaZach** [comment]
I plan to have this done before the next release (Aiming end of January or February)

**Comment by woogerboy21** [comment]
So would we have a setting in the clients settings that allow the user to choose the build types or will we work in some type of logic that checks for the most recent client available to both stable/unstable builds and offer the user the ability to install the one of there choosing?

**Comment by Daenyth** [comment]
User setting. Default value is full releases. User can pick from dropdown.
MVP version would be to have release + rc channels.
